kafka/pykafka/consumer.py

```diff
@@ -47,9 +199,63 @@ class OwnedPartition(object):
     Used to keep track of offsets and the internal message queue.
     """
 
-    def __init__(self, partition):
+    def __init__(self, partition, consumer):
         self.partition = partition
+        self.consumer = consumer
         self._messages = Queue()
+        self.last_offset_consumed = 0
+        self.next_offset = 0
+
+        if self.consumer._auto_commit_enable and self.consumer.consumer_group is not None:
+            self.last_offset_consumed = self._fetch_last_known_offset()
 
     def consume(self, timeout=None):
+        """Get a single message from this partition
+        """
+        if self._messages.empty():
+            self._fetch(timeout=timeout)
+
+        try:
+            message = self._messages.get_nowait()
+            self._queued_chunks -= 1
```

Comment:
Queued chunks as weird, and I think we want to avoid this setting.  My (admittedly poor) understanding of the Scala driver is that it has chunks of bytes queues and still serialized.  It then deserializes them on demand into messages and provides them to the user.  This seems odd to me because if you don't know the offset of the final message in a chunk (as it's still serialized), how do you know where to start fetching the next chunk?  It's all very confusing.

I talked with @yungchin about this a while back and IIRC librdkafka uses `max_messages_queued` which is simply the number of messages to keep in this per-partition queue.  It seems similar to how you've implemented `_queued_chunks`, so maybe all we need here is a name change.

@yungchin is my understanding of librdkafka here correct?  I remember us agreeing that the way there is better than the Scala driver